test-data/unit/check-selftype.test

```diff
@@ -1267,3 +1267,206 @@ class Test(Generic[T]):
         a: deque[List[T]]
         # previously this failed with 'Incompatible types in assignment (expression has type "deque[List[List[T]]]", variable has type "deque[List[T]]")'
         b: deque[List[T]] = a.copy()
+
+[case testTypingSelfBasic]
+from typing import Self, List
+
+class C:
+    attr: List[Self]
+    def meth(self) -> List[Self]: ...
```

Comment:
Can you also test returning `self` vs. returning `C()` inside a method that returns `Self` (the latter should be an error).